plugins/modules/ome_alert_policies.py

```diff
@@ -0,0 +1,940 @@
+            "value")) for y in x.get("parameters", []))) for x in inp_actions)
+        if 'Ignore' in inp_dict:
+            pld = {}
+            pld['TemplateId'] = ref_actions.get('Ignore').get('Id')
+            pld['Name'] = "Ignore"
+            pld['ParameterDetails'] = []
+            action_payload.append(pld)
+        else:
+            for inp_k, inp_val in inp_dict.items():
+                if inp_k in ref_actions:
+                    pld = {}
+                    if ref_actions.get(inp_k).get('Disabled'):
+                        module.exit_json(
+                            failed=True, msg=f"Action '{inp_k}' is disabled.")
+                    pld['TemplateId'] = ref_actions.get(inp_k).get('Id')
+                    pld['Name'] = inp_k
+                    diff = set(inp_val.keys()) - \
+                        set(ref_actions.get(inp_k).get('Parameters').keys())
+                    if diff:
+                        module.exit_json(
+                            failed=True, msg=f"Action '{inp_k}' has invalid parameters: {SEPARATOR.join(diff)}")
+                    pld['ParameterDetails'] = [
+                        {"Name": k, "Value": v} for k, v in inp_val.items()]
+                    action_payload.append(pld)
+                else:
+                    module.exit_json(
+                        failed=True, msg=f"Action '{inp_k}' does not exist.")
+        # module.exit_json(z=action_payload)
+    return {"Actions": action_payload}
+
+
+def get_category_or_message(module, rest_obj):
+    """
+    Retrieves a category or message based on the provided module and REST object.
+
+    Args:
+        module: The module containing the parameters for the API call.
+        rest_obj: The REST object used to retrieve the category data tree.
+
+    Returns:
+        cat_payload: The retrieved category or message payload.
+
+    Raises:
+        ExitJSON: If the category, sub-category or message does not exist.
+    """
+    cat_payload = {}
+    if module.params.get('category'):
+        inp_catalog_list = module.params.get('category')
+        cdict_ref = get_category_data_tree(rest_obj)
+        if not cdict_ref:
+            module.exit_json(failed=True, msg="Failed to fetch Category details.")
+        payload_cat_list = []
+        for inp_catalog in inp_catalog_list:
+            new_dict = {}
+            catalog_name = inp_catalog.get('catalog_name')
+            if catalog_name in cdict_ref:
+                new_dict["CatalogName"] = catalog_name
+                payload_cat = []
+                category_det = cdict_ref.get(catalog_name)
+                key_id = list(category_det.keys())[0]
+                payload_subcat = []
+                category_list = inp_catalog.get('catalog_category')
+                if category_list:
+                    for inp_category in category_list:
+                        if inp_category.get('category_name') in category_det:
```

Comment:
if possible, can we reduce line of code and have different small functions and avoid this many if else statement.

Reply:
cannot further reduce it, its necessary